New package: PRASInterface v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: PRASInterface
- Repository: https://github.com/NREL-Sienna/PRASInterface.jl
- Created by: @jd-lara
- Version: v0.1.0
- Commit: dcb927a69268e3a776172e4c1c5d8bc83bfadaf2
- Reviewed by: @jd-lara
- Reference: https://github.com/NREL-Sienna/PRASInterface.jl/commit/dcb927a69268e3a776172e4c1c5d8bc83bfadaf2#commitcomment-149447558
- Description: Interface to PRAS.jl maintained by Sienna\Ops
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd634d97cf51ab4d406eaeaef64696c3629214c62e428fce9033960f5ee6be5906a358728c08f6542f9827ec9bce36ae11c16331f19046829d23e2804a3ca8f5286dee9d8c76a2de9a5b34a571cf4d963696d9e9f07b9d92c3385044dd4068cacdad2411217a08ffc9829527d0b877959f6f184edbf3b571118b79dd2e6dfae3eae371af8cb6aa6870f92014f3714e41ddba13ad681d31a39a432af65174271e8adbaf41797945555209d342d72c671f27b -->

goerz commented:
Thank you for submitting your package! However, please make sure to add some documentation before registering. [At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), that would be a description of the package's purpose and a small usage example in the README.

It's not clear to me whether the name of the package is appropriate. I was not able to find the `PRAS.jl` referenced in the documentation. If it exists, `PRASInterface` seems like and unusual name (why would a Julia package need another Julia "Interface")?

goerz commented:
Update: I was now able to guess at the URL for the documentation https://nrel-sienna.github.io/PRASInterface.jl/dev/

Please note that the docs badge in the README is currently broken. That will probably resolve itself once this registration goes through and there is a tagged version, but please confirm that it works at that point.

You'll also want to add the "About" info from the documentation to the main README.

Now that I've seen the documentation, I would recommend the package name `PowerSystemsPRAS.jl`. The current name `PRASInterface` only makes sense from the perspective of a `PowerSystems` user. It would be appropriate in an [org-specific registry](https://github.com/GunnarFarneback/LocalRegistry.jl), or as a submodule name, but not for the wider audience of the General registry. There might be other orgs or packages that might want to interface with PRAS, after all. Moreover, since you already have `PowerSystems`, it is common practice and encouraged to use `PowerSystemsSomething` for "add-ons", where there is little scrutiny on the package name. You basically "own" that namespace.

jd-lara commented:
> Update: I was now able to guess at the URL for the documentation https://nrel-sienna.github.io/PRASInterface.jl/dev/
> 
> Please note that the docs badge in the README is currently broken. That will probably resolve itself once this registration goes through and there is a tagged version, but please confirm that it works at that point.
> 
> You'll also want to add the "About" info from the documentation to the main README.
> 
> Now that I've seen the documentation, I would recommend the package name `PowerSystemsPRAS.jl`. The current name `PRASInterface` only makes sense from the perspective of a `PowerSystems` user. It would be appropriate in an [org-specific registry](https://github.com/GunnarFarneback/LocalRegistry.jl), or as a submodule name, but not for the wider audience of the General registry. There might be other orgs or packages that might want to interface with PRAS, after all. Moreover, since you already have `PowerSystems`, it is common practice and encouraged to use `PowerSystemsSomething` for "add-ons", where there is little scrutiny on the package name. You basically "own" that namespace.

Thanks for the comments @goerz, on the name we would like to keep the existing one since it is an interface to this package https://github.com/NREL/PRAS that isn't own directly by us. This package is an interface between our data library PowerSystems.jl and PRAS that's where the names comes from. The difference is that the authors of PRAS decided not to register it in the general Julia registry. 

I will update the README and the registration request.

JuliaRegistrator committed:
New package: PRASInterface v0.1.0

UUID: 4b496c86-8d00-441d-b504-079c710e0aa7
Repo: https://github.com/NREL-Sienna/PRASInterface.jl.git
Tree: 76f8abe5d6a11c3067e295a2433d15393f3bc5a0

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

goerz commented:
Thanks for updating the README!

> on the name we would like to keep the existing one

[Like I said](https://github.com/JuliaRegistries/General/pull/120021#issuecomment-2495457836), I don't think the name `PRASInterface` is appropriate in the context of General. Is there something that you find objectionable about `PowerSystemsPRAS`?

Of course, I don't have any special authority, so you could also ask for other people's opinion on Slack. If there's a broad consensus in the community that the name is okay, I'll be happy to unblock.

jd-lara commented:
> Thanks for updating the README!
> 
> > on the name we would like to keep the existing one
> 
> [Like I said](https://github.com/JuliaRegistries/General/pull/120021#issuecomment-2495457836), I don't think the name `PRASInterface` is appropriate in the context of General. Is there something that you find objectionable about `PowerSystemsPRAS`?
> 
> Of course, I don't have any special authority, so you could also ask for other people's opinion on Slack. If there's a broad consensus in the community that the name is okay, I'll be happy to unblock.

The objection is that PowerSystemsPRAS doesn't reflect the true usage of this library, if we develop our own Probabilistic Resource Adequacy capability then there will be name confusion. This is not the correct package convention in our organization. 

The name we want to use is reflective of what the package actually does: an interface to PRAS.jl, similarly we registered in the past PowerModelsInterface.jl. It's an interface between our data model and another Julia modeling library period.  The difference is that in the case of PRAS.jl, a common library for users in the Resource Adequacy space, our team doesn't have the mechanism to make the authors register it to General. 

I don't understand the need to request community feedback for the registration of this package that follows the conventions of our organization that already have plenty of registered packages in the power systems modeling space.

adienes commented:
I think it is important to note that the name `PRAS.jl` would also likely be rejected from General. Recall rule 1 on the naming guidelines:

> 1. Avoid jargon. In particular, avoid acronyms unless there is minimal possibility of confusion.

so the fact that PRASInterface.jl is related to PRAS.jl is kind of irrelevant because PRAS.jl only exists in a private registry.

On another note, I'm wondering if this should be a [package extension](https://pkgdocs.julialang.org/v1/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)) rather than a standalone? it seems like the functionality here only exists in the context of PowerSystem.jl and PRAS.jl together, and it does not have any function as an independent package.

goerz commented:
> The name we want to use is reflective of what the package actually does: an interface to PRAS.jl

Maybe I'm confused. An interface between PRAS and what else? Whatever that "what else" is should be part of the package name. I thought this package was an Interface between `PowerSystems` and `PRAS`, hence the suggestions.

> similarly we registered in the past PowerModelsInterface.jl

At the time that was registered, there weren't as many volunteers reviewing package submissions. Thus, things slipped through the cracks. The name `PowerModelsInterface` has the same issue as `PRASInterface`: an interface is between two things, and the name does not indicate what those two things are. That package would probably not have been merged by today's standards.

> I don't understand the need to request community feedback

The General registry is a community resource, so all registrations will be reviewed

> for the registration of this package that follows the conventions of our organization

You are moving a package from the context of a specific organization to the General registry that addresses the entire community. For people *internal* to NREL-Sienna the name `PRASInterface` makes perfect sense. Someone in the general public does not have that context, so it is not clear what is interfacing with `PRAS` here. This wouldn't be a problem if packages in `Pkg` included the org name. That's something I'd love to have in Julia 2.0, but it's not what we have now. If this package is an interface between PRAS and general code inside your organization (rather than between a particular package like `PowerSystems` and `PRAS`), then probably `SiennaPRASInterface` would be the appropriate name.

goerz commented:
> PRAS.jl would also likely be rejected from General

Yes, or at least it would receive some pushback. I'm actually not sure how exactly `PRASInterface` manages to avoid an explicit dependency on `PRAS`. There appears to be some vendored code…

It does seem like a problem to have a registered package interfacing with `PRAS` when `PRAS` is not registered. There's a reason that dependencies of registered packages also have to be registered. If PRAS ever were to be registered, I can't guarantee that it would manage to get an exception from the naming guidelines, even though I'm open to it. If `PRAS` had to change names, `PRASInterface` would have a problem.

It would be good for @GordStephen to clarify what kind of plans there might be for a registration of `PRAS`, and/or whether NREL can guarantee that `PRAS` will exist as a permanent name.

Side node: if PRAS ever were registered, even if we make an exception for the name (which we very well might), the repository URL has to be changed to end with `.jl`. I'd recommend doing that anyway, as early as possible

jd-lara commented:
> > PRAS.jl would also likely be rejected from General
> 
> Yes, or at least it would receive some pushback. I'm actually not sure how exactly `PRASInterface` manages to avoid an explicit dependency on `PRAS`. There appears to be some vendored code…
> 
> It does seem like a problem to have a registered package interfacing with `PRAS` when `PRAS` is not registered. There's a reason that dependencies of registered packages also have to be registered. If PRAS ever were to be registered, I can't guarantee that it would manage to get an exception from the naming guidelines, even though I'm open to it. If `PRAS` had to change names, `PRASInterface` would have a problem.
> 
> It would be good for @GordStephen to clarify what kind of plans there might be for a registration of `PRAS`, and/or whether NREL can guarantee that `PRAS` will exist as a permanent name.
> 
> Side node: if PRAS ever were registered, even if we make an exception for the name (which we very well might), the repository URL has to be changed to end with `.jl`. I'd recommend doing that anyway, as early as possible

I can get behind using `SiennaPRASInterface` if it moves the conversation forward. Please just confirm so I can change the name.

goerz commented:
@rayegun Point taken. That phrasing was indeed out of line. I apologize.

goerz commented:
We talked this out on Slack. @jd-lara will check to see if renaming to `SierraPRASInterface` is easy to do. If the internal procedures within NREL make a rename too cumbersome (we're talking about a government agency, after all), we'll just stick to `PRASInterface` and I'll unblock.

jd-lara commented:
This PR can be closed given https://github.com/JuliaRegistries/General/pull/120172
